huntsman/camera/pyro.py

```diff
@@ -327,44 +307,6 @@ def autofocus(self,
         assert self.focuser.is_connected, \
             self.logger.error("Focuser must be connected for autofocus.")
 
-        # Make certain that all the argument are builtin types for easy Pyro serialisation
-        if isinstance(seconds, u.Quantity):
-            seconds = seconds.to(u.second).value
-        if seconds is not None:
-            seconds = float(seconds)
-
-        if focus_range is not None:
-            focus_range = (int(limit) for limit in focus_range)
-
-        if focus_step is not None:
-            focus_step = (int(step) for step in focus_step)
-
-        if keep_files is not None:
-            keep_files = bool(keep_files)
-
-        if take_dark is not None:
-            take_dark = bool(take_dark)
-
-        if thumbnail_size is not None:
-            thumbnail_size = int(thumbnail_size)
-
-        merit_function = str(merit_function)
-        merit_function_kwargs = dict(merit_function_kwargs)
-
-        if mask_dilations is not None:
-            mask_dilations = int(mask_dilations)
-
-        if coarse is not None:
-            coarse = bool(coarse)
-
-        if make_plots is not None:
-            make_plots = bool(make_plots)
-
-        if isinstance(timeout, u.Quantity):
-            timeout = timeout.to(u.second).value
-        if timeout is not None:
-            timeout = float(timeout)
-
```

Comment:
Just curious as to why we are able to remove all this now if was needed earlier.

Reply:
The quantities stuff is now redundant because of the quantity (de)serialiser. The other input validation isn't needed because the remote method call will raise exceptions for these, which will be re-raised locally. I think these were originally added either because of an over abundance of caution, or because I couldn't get the remote exception hook working properly. Either way, not needed now.